metabase/metabase
# feat(sdk): connect databases, generate models and x-rays from the CLI

Closes #46412 and #46416 

### Description

To make onboarding for the embedding sdk smoother, we allow the customers to add a database connection and pick a couple of tables from within the CLI, which will create models and x-ray dashboards automatically for them. This data can be used when generating example dashboard components.

### How to verify

- Run `yarn build-embedding-sdk:watch` in a terminal, then `node resources/embedding-sdk/dist/cli.js start` in another terminal
- Go through the steps - try to add a database connection, and pick a couple of tables. See the demo video below.
- You should see a couple of generated models and x-ray dashboards.

### Demo

<div>
    <a href="https://www.loom.com/share/c8b2b692d4a446838b60c0b7e356192b">
      <p>August 6 - CLI database, model & x-ray demo (WIP) - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/c8b2b692d4a446838b60c0b7e356192b">
      <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/c8b2b692d4a446838b60c0b7e356192b-3e31cc30876cbcd2-full-play.gif">
    </a>
  </div>
